source/DataElement/AMDSFlatArray.cpp

```diff
@@ -718,38 +718,41 @@ bool AMDSFlatArray::resetTargetArrayAndReplaceData(AMDSFlatArray *targetArray) c
 }
 
 void AMDSFlatArray::resizeType(AMDSDataTypeDefinitions::DataType dataType, quint32 size){
+	if( (dataType_ == dataType) && (size == this->size()) )
+		return;
+
 	dataType_ = dataType;
```

Comment:
When the dataType is changed, do we need to clear the current buffer before we lost the original dataType?